### PR TITLE
Form

### LINKED ISSUE #31: React Hook Form
# User-Story

As a developer

I want to implement react hook form

so that i can more easily validate my form

# Notes, Links, Material

https://react-hook-form.com/

# ToDos

- [x] Install react hook form
- [x] Implement in existing form

### PR DESCRIPTION
**Pull request by kschmitz123**
I've completed my form element and implemented react-hook-form.

So far you can upload images to cloudinary. 
ToDo: Add database so that text can be uploaded as well.

#close #31

### CODE REVIEW

File: client/src/components/Header.js [hunk 1]
```diff
@@ -38,9 +38,7 @@ const Title = styled.h2`
 
 export const Header = ({ title }) => {
   const location = useLocation();
-  useEffect(() => {
-    console.log(location.pathname);
-  }, [location]);
+  useEffect(() => {}, [location]);
```
Comment (Sina-Sche) [inline client/src/components/Header.js:41]: nice use of UseEffect/useLocation ! 💯

**Review by Sina-Sche — APPROVED** [review]
Wow! Very good job 🚀 💯

File: client/src/components/UploadProject.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { Button } from "./Button";
```
Comment (bwnstck) [inline client/src/components/UploadProject.js:2]: you should use export default for Components

like here :) https://github.com/DiscoDevs/DispoDisco/pull/76

File: client/src/components/InputElements.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  required: true,
+};
+
+LargeInput.defaultProps = {
+  maxLength: "500",
+  required: true,
+  type: "text",
+  placeholder: "Enter project description (max 500 characters)",
+};
+
```
Comment (bwnstck) [inline client/src/components/InputElements.js:58]: I think it is common to have each Component in a separate file (and then default export) 🦄

File: client/src/components/UploadProject.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  const uploadImage = async (base64EncodedImage) => {
+    try {
+      await fetch("/api/upload", {
+        method: "POST",
+        body: JSON.stringify({ data: base64EncodedImage }),
+        headers: { "Content-Type": "application/json" },
+      });
+      alert("Image successfully uploaded.");
```
Comment (bwnstck) [inline client/src/components/UploadProject.js:46]: 👀 Do you really want this ?

File: client/src/pages/AddProjectPage.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 import React from "react";
 import styled from "styled-components/macro";
 import { Header } from "../components/Header";
-import { Input } from "../components/Input";
 import { Navbar } from "../components/Navbar";
```
Comment (bwnstck) [inline client/src/pages/AddProjectPage.js:4]: default export like in the comment before 👍

File: server.js [hunk 6]
```diff
@@ -13,8 +13,8 @@ app.post("/api/upload", async (request, response) => {
     const uploadedResponse = await cloudinary.uploader.upload(fileStr, {
       upload_preset: "upload",
     });
-    console.log(uploadedResponse);
-    response.status(200).send(uploadedResponse);
+    console.log(uploadedResponse.secure_url);
```
Comment (bwnstck) [inline server.js:16]: do you need this one? secure.url looks like something you dont want to log in the console 👩‍🎤

**Review by bwnstck — CHANGES_REQUESTED** [review]

**Comment by bwnstck** [comment]
for the rest 💯 🤘 🚀

File: server.js (hunk 6 above)
Comment (kschmitz123) [reply server.js:16]: i need this for later . i console logged it so i dont forget it 🙈

File: client/src/components/UploadProject.js (hunk 4 above)
Comment (kschmitz123) [reply client/src/components/UploadProject.js:46]: the alert? for now yes . i'll focus on the functionality first and after that i'll implement nicer messages 👩🏻‍🎨

File: server.js (hunk 6 above)
Comment (bwnstck) [reply server.js:16]: nagut 🙃

File: client/src/components/UploadProject.js (hunk 4 above)
Comment (bwnstck) [reply client/src/components/UploadProject.js:46]: okey accepted 😉

**Pull request merged**